### ISSUE TITLE
GPU Operator crashes if no GPU nodes present

### ISSUE THREAD

**Issue by mmgaggle** [NVIDIA/gpu-operator#71]
### 1. Issue or feature description

GPU Operator crashes when GPU nodes are not present, as is the case when GPU nodes are temporarily scaled down to save operational costs. By extension, this means that a GPU node must be idling while the NVIDIA kernel module is built.

### 2. Steps to reproduce the issue

* Create machineset using GPU instance type
* Scale replica for said machineset to 1
* Install GPU operator, create ClusterPolicy (a truly ominous CRD name)
* Scale replica for machineset to 0
* `oc get po -n gpu-operator-resources`

### 3. More Information

https://gist.github.com/mmgaggle/27a847e0ed8a7a347dfd026471c67a9b

**Comment by mmgaggle** [comment]
Perhaps the culprit?

https://github.com/NVIDIA/gpu-operator/blob/master/assets/state-driver/0500_daemonset.yaml#L57-L58

**Comment by jmvizcainoio** [comment]
Hi,

I'm using cluster autoscaler to add gpu instances dynamically and i have the same problem, gpu-operator crash when doesn't exists gpu nodes.

```
{"level":"info","ts":1604328402.193679,"logger":"controller_clusterpolicy","msg":"Could not get any nodes to match nvidia.com/gpu.present label","Request.Namespace":"default","Request.Name":"Node","ERROR":""}
{"level":"info","ts":1604328402.1937275,"logger":"controller_clusterpolicy","msg":"Failed to apply transformation 'nvidia-driver-daemonset' with error: 'ERROR: Could not find kernel full version: ('', '')'"}
```

Thanks

**Comment by shivamerla** [comment]
@mmgaggle @jmvizcainoio Yes, gpu-operator needs at-least one gpu node to function. This is due to check [here](https://github.com/NVIDIA/gpu-operator/blob/master/pkg/controller/clusterpolicy/object_controls.go#L232). This has been the behavior with all previous versions as well. If you are not planning to add gpu-nodes, i would recommend scaling the operator deployment pod to 0 and scale it back when gpu node is present to prevent un-install and install again. We will look into handling this case in the future.

**Comment by shivamerla** [comment]
This is fixed with 1.5.0 release.
